internal/monitors/manager.go

```diff
@@ -314,26 +314,49 @@ func (mm *MonitorManager) findConfigForMonitorAndRun(endpoint services.Endpoint)
 // endpoint may be nil for static monitors
 func (mm *MonitorManager) createAndConfigureNewMonitor(config config.MonitorCustomConfig, endpoint services.Endpoint) error {
 	id := types.MonitorID(mm.idGenerator())
+	coreConfig := config.MonitorConfigCore()
+	monitorType := coreConfig.Type
 
 	log.WithFields(log.Fields{
-		"monitorType":   config.MonitorConfigCore().Type,
-		"discoveryRule": config.MonitorConfigCore().DiscoveryRule,
+		"monitorType":   monitorType,
+		"discoveryRule": coreConfig.DiscoveryRule,
 		"monitorID":     id,
 	}).Info("Creating new monitor")
 
 	instance := newMonitor(config.MonitorConfigCore().Type, id)
 	if instance == nil {
-		return errors.Errorf("Could not create new monitor of type %s", config.MonitorConfigCore().Type)
+		return errors.Errorf("Could not create new monitor of type %s", monitorType)
+	}
+
+	metadata, ok := MonitorMetadatas[monitorType]
+	if !ok {
+		panic(fmt.Sprintf("could not find monitor metadata of type %s", monitorType))
 	}
 
 	configHash := config.MonitorConfigCore().Hash()
 
+	var extraMetrics []string
+
+	// Monitors can add additional extra metrics to allow through such as based on config flags.
+	if monitorExtra := config.GetExtraMetrics(); monitorExtra != nil {
+		extraMetrics = append(extraMetrics, coreConfig.ExtraMetrics...)
+		extraMetrics = append(extraMetrics, monitorExtra...)
+	} else {
+		extraMetrics = coreConfig.ExtraMetrics
+	}
+
+	extraMetricsFilter, err := newMetricsFilter(metadata, extraMetrics, coreConfig.ExtraGroups)
+	if err != nil {
+		return fmt.Errorf("unable to construct extraMetrics filter: %s", err)
+	}
+
 	output := &monitorOutput{
-		monitorType:               config.MonitorConfigCore().Type,
+		monitorType:               monitorType,
 		monitorID:                 id,
-		notHostSpecific:           config.MonitorConfigCore().DisableHostDimensions,
-		disableEndpointDimensions: config.MonitorConfigCore().DisableEndpointDimensions,
-		filter:                    config.MonitorConfigCore().Filter,
+		notHostSpecific:           coreConfig.DisableHostDimensions,
+		disableEndpointDimensions: coreConfig.DisableEndpointDimensions,
+		filter:                    coreConfig.Filter,
```

Comment:
I apologize in advance but I'm going to make a merge conflict with this once I merge https://github.com/signalfx/signalfx-agent/pull/701.

Is it possible to unify the `extraMetricsFilter` with the other filters by making it implement `dpfilters.DatapointFilter` interface?

Reply:
Yeah would make sense to unify them. The one thing I like about the name `shouldSend` is that it makes it clear what the boolean return value means. With `Matches` I forget if `true` means it will be sent or won't be sent. Having the interface be `ShouldSend() bool` would make this clear but also seems kind of weird. Thoughts?

Reply:
As we talked about, the filter variable that is being used to call `Matches` on should be named something descriptive that indicates whether it drops or includes things.